### PR TITLE
Documentation: discovery-fallback flag to proxy.md

### PR DESCRIPTION
**Pull request by gyuho**
And partially related to https://github.com/coreos/etcd/issues/4595.

### CODE REVIEW

**Comment by xiang90** [comment]
I think we need to change line 69.

File: Documentation/proxy.md [hunk 1]
```diff
@@ -133,10 +133,11 @@ $ export ETCD_NAME="infra1"
...
 $ export ETCD_INITIAL_CLUSTER_STATE=existing
 $ etcd -listen-client-urls http://10.0.1.11:2379 \
--advertise-client-urls http://10.0.1.11:2379 \
--listen-peer-urls http://10.0.1.11:2380 \
--initial-advertise-peer-urls http://10.0.1.11:2380 \
--data-dir %data_dir%
+--advertise-client-urls http://10.0.1.11:2379 \
+--listen-peer-urls http://10.0.1.11:2380 \
+--initial-advertise-peer-urls http://10.0.1.11:2380 \
+--discovery-fallback 'proxy'
```
Comment (xiang90) [inline Documentation/proxy.md]: why add this flag?

Comment (gyuho) [inline Documentation/proxy.md]: Right. This is wrong place. Will move. Thanks.

File: Documentation/proxy.md [hunk 2]
```diff
@@ -49,19 +49,20 @@ To start a proxy that will connect to a statically defined etcd cluster, specify
...
--initial-cluster infra0=http://10.0.1.10:2380,infra1=http://10.0.1.11:2380,infra2=http://10.0.1.12:2380
+--listen-client-urls http://127.0.0.1:8080 \
+--initial-cluster infra0=http://10.0.1.10:2380,infra1=http://10.0.1.11:2380,infra2=http://10.0.1.12:2380
 ```
 
 ### Start an etcd proxy with the discovery service
 If you bootstrap an etcd cluster using the [discovery service][discovery-service], you can also start the proxy with the same `discovery`.
 
-To start a proxy using the discovery service, specify the `discovery` flag. The proxy will wait until the etcd cluster defined at the `discovery` url finishes bootstrapping, and then start to forward the requests.
+To start a proxy using the discovery service, specify the `discovery` flag. The proxy will wait until the etcd cluster defined at the `discovery` url finishes bootstrapping, and then start to forward the requests. Specify `discovery-fallback` flag to make extra members fall back to proxy when the cluster is full.
```
Comment (gyuho) [inline Documentation/proxy.md]: Just added this line (at the end). PTAL.

/cc @xiang90 

Thanks.

File: Documentation/proxy.md [hunk 3]
```diff
@@ -60,13 +60,14 @@ To start a proxy using the discovery service, specify the `discovery` flag. The
 
 ```
 etcd -proxy on \
--listen-client-urls http://127.0.0.1:8080 \
--discovery https://discovery.etcd.io/3e86b59982e49066c5d813af1c2e2579cbf573de
+--listen-client-urls http://127.0.0.1:8080 \
+--discovery https://discovery.etcd.io/3e86b59982e49066c5d813af1c2e2579cbf573de \
+--discovery-fallback 'proxy'
```
Comment (xiang90) [inline Documentation/proxy.md]: why do you add this flag here?

File: Documentation/proxy.md [hunk 4]
```diff
@@ -60,13 +60,14 @@ To start a proxy using the discovery service, specify the `discovery` flag. The
...
 ```
 
 ## Fallback to proxy mode with discovery service
 
-If you bootstrap an etcd cluster using [discovery service][discovery-service] with more than the expected number of etcd members, the extra etcd processes will fall back to being `readwrite` proxies by default. They will forward the requests to the cluster as described above. For example, if you create a discovery url with `size=5`, and start ten etcd processes using that same discovery url, the result will be a cluster with five etcd members and five proxies. Note that this behaviour can be disabled with the `proxy-fallback` flag.
+If you bootstrap an etcd cluster using [discovery service][discovery-service] with more than the expected number of etcd members, the extra etcd processes will fall back to being `readwrite` proxies by default. They will forward the requests to the cluster as described above. For example, if you create a discovery url with `size=5`, and start ten etcd processes using that same discovery url, the result will be a cluster with five etcd members and five proxies. Note that this behaviour can be disabled with the `proxy-fallback='exit'` flag.
```
Comment (xiang90) [inline Documentation/proxy.md]: As I mentioned, we should change proxy-fallback to discovery-fallback. Please check our flagset.

Comment (gyuho) [inline Documentation/proxy.md]: Ah sorry, missed the typo. Will fix.

File: Documentation/proxy.md (hunk 3 above)
Comment (gyuho) [inline Documentation/proxy.md]: Not needed here. Will remove. Thanks.

**Comment by xiang90** [comment]
lgtm

**Comment by gyuho** [comment]
thanks.

**Pull request merged**